stream-video-android-core/src/main/kotlin/io/getstream/video/android/core/StreamVideoBuilder.kt

```diff
@@ -235,6 +235,9 @@ public class StreamVideoBuilder @JvmOverloads constructor(
             scope.launch {
                 try {
                     val result = client.connectAsync().await()
+                    if (notificationConfig.autoRegisterPushDevice) {
```

Comment:
I remember I moved this to block that runs when we get the Connected event for some reason. I'll check tomorrow and run a test.

Reply:
Checked this. It looks good because `connectAsync` is `await`ed and we don't have that `continuation` inside it anymore. The problem was that `createDevice` sends the token to the backend, but it needs to have a user associated (i.e. connected) with it to work.